### PR TITLE
Comment thumbnails

### PR DESCRIPTION
**Pull request by osilkin98**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes 
Adds comment thumbnails

Issue Number: #3107

## What is the current behavior?
No thumbnails on comments

## What is the new behavior?
Thumbnails on comments & `ChannelThumbnail` allows for default, anonymous thumbnails now if nothing is passed in.

### CODE REVIEW

**Comment by seanyesmunt** [comment]
@osilkin98 Have you noticed any slowness issues with resolving a lot of channels when a file has a lot of comments?

Could you try batching all the resolves to see how that affects things? You could call resolve from `CommentList` and pass all the uris to that.

**Comment by osilkin98** [comment]
Currently, the styling is as it should be for the most part. It appears that the length of the comment list will cause slowdown issues by `FilePage` re-rendering on scroll. The two possible solutions it seems are to either implement `FilePage.shouldComponentUpdate`, or by converting either `FilePage` or `CommentList` into inheriting from `PureComponent`

**Comment by osilkin98** [comment]
Another solution could be to virtualize the comment using `react-window` or a similar library.

**Comment by seanyesmunt** [comment]
Sorry I missed your comment @osilkin98 

What component is causing that?

**Comment by osilkin98** [comment]
The `CollapseableText` component seems to be redrawing itself frequently,
though I don't know if that's the sole cause.

File: ui/component/channelThumbnail/view.jsx [hunk 1]
```diff
@@ -6,18 +6,23 @@ import Gerbil from './gerbil.png';
...
   // Generate a random color class based on the first letter of the channel name
-  const { channelName } = parseURI(uri);
-  const initializer = channelName.charCodeAt(0) - 65; // will be between 0 and 57
+  let initializer;
+  if (thumbnail) {
+    const { channelName } = parseURI(uri);
+    initializer = channelName.charCodeAt(0) - 65; // will be between 0 and 57
+  } else {
+    // if we want to default a thumbnail
+    initializer = Math.floor(Math.random() * 104729); // 10000th prime number
```
Comment (seanyesmunt) [inline ui/component/channelThumbnail/view.jsx]: Why did you change this?

Comment (osilkin98) [reply ui/component/channelThumbnail/view.jsx]: You need some way of generating the thumbnails on anonymous comments & claims.
